### DISCUSSION TITLE
how to use 'Exponential Moving Average' for weights ?

### DISCUSSION THREAD

**Discussion by IamMegatron2025** [apache/incubator-mxnet#17122] [Q&A] [answered]
Hello, I cannot find some materials about how to use 'Exponential Moving Average' for weights by python API. Can you give some advises?

**Comment by sxjscience** [comment]
@IvyGongoogle Would you provide more information? Are you referring to something similar to `tf.train.ExponentialMovingAverage`?

**Comment by szha** [comment, marked as answer]
Assuming that's the case, we have a prototype in GluonNLP for this at: https://github.com/dmlc/gluon-nlp/blob/qanet/scripts/question_answering/util.py#L74

**Comment by IamMegatron2025** [comment]
@sxjscience  yes. I mean that.  @szha  Could you provide some `py` scripts about how to use this class `ExponentialMovingAverage` when I am defining my cnn model ?
